code/game/objects/items/devices/radio/headset.dm

```diff
@@ -358,23 +358,23 @@ GLOBAL_LIST_INIT(channel_tokens, list(
 
 /obj/item/radio/headset/screwdriver_act(mob/living/user, obj/item/tool)
 	user.set_machine(src)
-	if(keyslot || keyslot2)
+	if(keyslot && keyslot.canRemove || keyslot2 && keyslot2.canRemove)
 		for(var/ch_name in channels)
 			SSradio.remove_object(src, GLOB.radiochannels[ch_name])
 			secure_radio_connections[ch_name] = null
 
-		if(keyslot)
+		if(keyslot && keyslot.canRemove)
 			user.put_in_hands(keyslot)
 			keyslot = null
-		if(keyslot2)
+		if( keyslot2 && keyslot2.canRemove)
 			user.put_in_hands(keyslot2)
 			keyslot2 = null
 
 		recalculateChannels()
 		to_chat(user, span_notice("You pop out the encryption keys in the headset."))
-
 	else
-		to_chat(user, span_warning("This headset doesn't have any unique encryption keys! How useless..."))
+		to_chat(user, span_warning("This headset doesn't have any encryption keys!"))
```

Comment:
Should include a detail if there are keys, just un-removable ones (Because I think they still show up on examine? So it might be misleading) -
```suggestion
		to_chat(user, span_warning("This headset doesn't have any[keyslot || keyslot2 ? " removable" : ""] encryption keys!"))
```